src/commands/app/db/status.js

```diff
@@ -0,0 +1,198 @@
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { DBBaseCommand } from '../../../DBBaseCommand.js'
+import { Flags } from '@oclif/core'
+import chalk from 'chalk'
+
+export class Status extends DBBaseCommand {
+  async run () {
+    const { watch } = this.flags
+
+    this.debugLogger?.info?.('Checking database provisioning status')
+
+    if (watch) {
+      return this.watchStatus()
+    } else {
+      return this.checkStatus()
+    }
+  }
+
+  async checkStatus () {
+    try {
+      this.log(chalk.blue('Checking database provisioning status...'))
+
+      const provisionStatusResponse = await this.db.provisionStatus()
+      this.debugLogger?.info?.('Status result:', provisionStatusResponse)
+
+      this.displayStatus(provisionStatusResponse)
+
+      const result = {
+        ...provisionStatusResponse,
+        namespace: this.rtNamespace,
+        timestamp: new Date().toISOString()
+      }
+
+      return result
+    } catch (error) {
+      this.debugLogger?.error?.('Status command error:', error)
+
+      if (error.message.includes('not found') || error.message.includes('404')) {
+        this.log(chalk.yellow('No database has been provisioned for this workspace'))
+        this.log(chalk.dim(`   Namespace: ${this.rtNamespace}`))
+        this.log(chalk.dim('   Status: NOT_PROVISIONED'))
+
+        return {
+          status: 'NOT_PROVISIONED',
+          namespace: this.rtNamespace,
+          timestamp: new Date().toISOString()
+        }
+      }
+
+      this.log(chalk.red('Failed to check database status'))
+      this.log(chalk.dim(`   Error: ${error.message}`))
+
+      return {
+        status: 'error',
+        namespace: this.rtNamespace,
+        error: error.message,
+        timestamp: new Date().toISOString()
+      }
+    }
+  }
+
+  async watchStatus () {
+    this.log(chalk.blue('Watching database provisioning status (press Ctrl+C to stop)...'))
+
+    let previousStatus = null
+    const checkInterval = 3000 // 3 seconds
+
+    const watchLoop = async () => {
+      try {
+        const provisionStatusResponse = await this.db.provisionStatus()
+
+        // Only display if status changed
+        if (!previousStatus || previousStatus.status !== provisionStatusResponse.status) {
+          this.log(chalk.dim(`\n[${new Date().toLocaleTimeString()}]`))
+          this.displayStatus(provisionStatusResponse, false) // Don't show timestamp in watch mode
+          previousStatus = provisionStatusResponse
+
+          // Stop watching if provisioning is complete or failed
+          const currentStatus = provisionStatusResponse.status.toUpperCase()
+          if (currentStatus === 'PROVISIONED' || currentStatus === 'FAILED') {
+            this.log(chalk.dim('\nProvisioning completed. Stopping watch mode.'))
```

Comment:
```suggestion
            this.log(chalk.dim('\nStopping watch mode.'))
```

A failure state has not had provisioning completed, but still stop the watch mode.